WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -71,27 +91,55 @@ - (BOOL)fb_typeText:(NSString *)text frequency:(NSUInteger)frequency error:(NSEr
 
 - (BOOL)fb_clearTextWithError:(NSError **)error
 {
-  if (0 == [self.value fb_visualLength]) {
+  id value = self.value;
+  if (nil != value && ![value isKindOfClass:NSString.class]) {
+    return [[[FBErrorBuilder builder]
+               withDescriptionFormat:@"The value of '%@' element is not a string and thus cannot be cleared", self.description]
+              buildError:error];
+  }
+  
+  if (nil == value || 0 == [value fb_visualLength]) {
+    // Short circuit if the content is not present
     return YES;
   }
 
   if (![self fb_prepareForTextInputWithError:error]) {
     return NO;
   }
   
-  NSUInteger preClearTextLength = 0;
-  NSData *encodedSequence = [@"\\u0008\\u007F" dataUsingEncoding:NSASCIIStringEncoding];
-  NSString *backspaceDeleteSequence = [[NSString alloc] initWithData:encodedSequence encoding:NSNonLossyASCIIStringEncoding];
-  while ([self.value fb_visualLength] != preClearTextLength) {
-    NSMutableString *textToType = @"".mutableCopy;
-    preClearTextLength = [self.value fb_visualLength];
-    for (NSUInteger i = 0 ; i < preClearTextLength ; i++) {
-      [textToType appendString:backspaceDeleteSequence];
+  static NSString *backspaceDeleteSequence;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    backspaceDeleteSequence = [[NSString alloc] initWithData:(NSData *)[@"\\u0008\\u007F" dataUsingEncoding:NSASCIIStringEncoding]
+                                                    encoding:NSNonLossyASCIIStringEncoding];
+  });
+  
+  NSUInteger retry = 0;
+  NSUInteger preClearTextLength = [value fb_visualLength];
+  do {
+    NSString *textToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
+    if (retry >= MAX_CLEAR_RETRIES - 1) {
+      // Last chance retry. Try to select the content of the field using the context menu
+      [self pressForDuration:1.0];
+      XCUIElement *selectAll = self.application.menuItems[@"Select All"];
```

Comment:
The name is only for English. We can also find it as the 2nd item by `XCUIElementTypeMenuItem`, but this way is also not so stable way

Reply:
yep, we cannot cover all possible cases there, but this is anyway better than nothing